### DISCUSSION TITLE
No documentation for auth apis?

### DISCUSSION THREAD

**Discussion by dhruvbhatia7** [supabase/supabase#3344] [Questions] [answered]
I'm trying to find documentation for auth apis, specifically documentation for stuff like:

`supabase.auth.api.setAuthCookie(req, res)`

This seems to be super useful since I'm using nextjs, unfortunately I couldn't find documentation for auth.api functions

**Comment by soedirgo** [comment, marked as answer]
Hey @dhruvbhatia7, you can find the complete documentation here: https://supabase.github.io/gotrue-js/classes/GoTrueApi.html#setAuthCookie. It's autogenerated though, so it's a bit messy.

**Comment by cprcrack** [comment]
I'm in the same situation. Wanted to use authentication from the server, and there are mentions in the docs [here](https://supabase.com/docs/reference/javascript/auth-user) to the use of `auth.api.getUserByCookie()` but no docs for auth.api themselves. And the link provided by @soedirgo no longer works, what would be the link now?

And, is authentication from the server even supported? I want to take advantage of supabase auth features but don't want any supabase code on my client, want to handle it from an express backend.

**Reply by kk21** [reply]
Search ard the discussions here for supabase.auth.api.getUser(<user's jwt token>) to check if the jwt is valid (not expired), this authenticates the user from your own server.

**Reply by soedirgo** [reply]
Hey @cprcrack, I've updated the link above if it helps :+1:
